.NET Core support for .NET Standard 1.6

eloekset opened the pull request:
Now that VS2017 is soon being released (March 7) I've done another attempt at adding support for .NET Standard.

I've based this work on @alski's work PR #11.

I've added a new solution file for .NET Core to not break the current solution, and also I couldn't build the old solution on my computer using VS2017.

I hope we can have a public preview of NuGet packages with .NET Standard support, since my work on porting SharpRepository to .NET Core depends on Should.

Commit:
Many project flavours now added. All compiling including UWP where some type calls need additional GetTypeInfo.

Commit:
Move to Shared project model

Commit:
Update tests projects

Commit:
Use commonAssemblyInfo in UWP

Commit:
Update nant

Commit:
Fail to get Nant to build and package WUP

Commit:
Move to NuGet Packaging
(https://visualstudiogallery.msdn.microsoft.com/daf5c6db-386b-4994-bdd7-b6cd52f11b72?SRC=VSIDE).
Still needs netcore system dependencies adding

Commit:
Added NuGet packaging

Commit:
Get cmd line build working and tidy up

Commit:
Fix FLuent Package
Why didn't this commit last time? Hmmm.

alski committed:
Make compatible with Nuget v3

alski committed:
Update README.markdown

Commit:
Attempt to get building on MyGet

Commit:
Minimal 3.5 only build for myget

Commit:
Even more cut down

Commit:
Seriously!

Commit:
Removing push to nuget.org

Commit:
Should be a working v2.5 only build and nupack

Commit:
Try adding net40 back in

Commit:
Alternate packaging method for MyGet

Commit:
Remove packager from build

Commit:
Can myget build UWP?

Commit:
additional restore

Commit:
Try moving to more up to date libraries

Commit:
Upgrade to VS2015 update3

Commit:
Re-enable Packaging

Commit:
Rename nuspecs back to match powershell

Commit:
A working MyGet compatible build

:-)

Commit:
Tidy up a comment so I get another version to build!

Commit:
Temporarily disabling package

Commit:
Remove publishing and try to insert version number

Commit:
Remove packaging

Commit:
Working build AND test

Commit:
Use simple packaging

Commit:
Pack with a version number and handle fallback

Commit:
Consolidate MSBuildTasks version

eloekset committed:
Add projects that target .NET Standard 1.6.

eloekset committed:
Upgrade nuget.exe to 3.5.0 for .NET Standard support.

eloekset committed:
Add target netstandard1.6 to NuGet packages.

eloekset committed:
Upgrade build script to build DotNetCore solution.

alski commented:
@eloekset Eivind this looks to be excellent. I really hope @erichexter can pick this up soon.

In the mean time, I'll update the feed [![shoulduwp MyGet Build Status](https://www.myget.org/BuildSource/Badge/shoulduwp?identifier=b8cd0611-0efd-4271-b774-496f62f2c644)](https://www.myget.org/) https://www.myget.org/feed/shoulduwp/package/nuget/Should so that we avoid creating too many forks :-)

NuGet as https://www.myget.org/F/shoulduwp/api/v3/index.json 

eloekset committed:
Add dependencies to NuGet packages for .NET Standard 1.6.

Commit:
Merge remote-tracking branch 'refs/remotes/erichexter/master'

eloekset commented:
@alski, I rebased on your master now in case any of the latest changes has impacts on the build. However I'm still not able to build.

`
error MSB4019: The imported project "C:\GitHub\eloekset\Should\.build\MSBuild
Tasks.1.5.0.214\tools\MSBuild.Community.Tasks.Targets" was not found. Confirm that the path in the <Import> declaration
 is correct, and that the file exists on disk.
`

Also I updated the nuspec files to add dependency on System.Runtime.Serialization.Primitives for the netstandard1.6 target.

alski commented:
I was just looking at this, it appears that the myget build agrees with you. 
https://www.myget.org/BuildSource/List/shoulduwp#d1848c26-95ca-4a0f-aa96-39c3bccc97dd

Just my machine that's the odd one out.

alski commented:
Found it.

build.bat pulls the latest MSbuildTasks package from http://www.nuget.org/packages/MSBuildTasks/ which has recently updated from 1.5.0.214

ms.build then uses 
`<PropertyGroup>
    <MSBuildCommunityTasksPath>\.build\MSBuildTasks.1.5.0.214\tools\</MSBuildCommunityTasksPath>
  </PropertyGroup>
  <Import Project=".build\MSBuildTasks.1.5.0.214\tools\MSBuild.Community.Tasks.Targets" />`    

What we need is something that detects what the version was, and injects it into ms.build, probably via an environment var or similar. e.g.
`<PropertyGroup>
    <MSBuildCommunityTasksPath>\.build\MSBuildTasks.`**`$(MSBuildToolsVersion)`**`\tools\</MSBuildCommunityTasksPath>
  </PropertyGroup>
  <Import Project=".build\`**`$(MSBuildVersion)`**`\tools\MSBuild.Community.Tasks.Targets" />`    

I'd be surprised if this doesn't already exist, so just need to find out how to do it.

For now I've hardcoded build.bat with
tools\nuget.exe install MSBuildTasks -o .build\ **-version 1.5.0.214**

alski commented:
Nope more, errors still working on it.

erichexter commented:
Last time I pulled the branch it would from a command line, so I punted on
it. If you can work out then dependencies and get them all in nuget
packages I'll accept the pull request.


On Sun, Feb 26, 2017 at 6:55 AM Al <notifications@github.com> wrote:

> Nope more, errors still working on it.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/erichexter/Should/pull/14#issuecomment-282553934>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AAHABN8rhELCyuElSRb1L1Qhr6_nCOn3ks5rgXZKgaJpZM4MMJ9b>
> .
>
-- 
Eric Hexter

blog | http://Hex.LosTechies.com
info | http://www.linkedin.com/in/erichexter

eloekset commented:
The commits in this PR are already merged into the original PR. All changes here will get merged as soon as the build is fixed, so I'll close this one now.

alski commented:
Hi eric, Did you ever look at the fix I put in for that? 
https://github.com/erichexter/Should/pull/11/commits/8fd5ed74e3ece7960e43795816ee7276faafe223
If that didn't fix the problem can you drop another gist of what is going wrong?

Also, can someone try another build please, as I'm seeing massively different issues on the myget platform which may not support the VS17 builds,
`C:\Users\Wonka\.nuget\packages\microsoft.net.native.compiler\1.6.0\build\Microsoft.Net.Native.Compiler.targets(37,5): error : The Microsoft.Net.Native.Compiler package, included in Microsoft.NETCore.UniversalWindowsPlatform 5.3 and later, is only supported in Visual Studio 2017 and later. You can downgrade the version of the Microsoft.NETCore.UniversalWindowsPlatorm NuGet package to 5.2.* or earlier for Visual Studio 2015. [D:\temp\f4caf7e\src\Should.Fluent-netcore45\Should.Fluent-netCore45.csproj]`

as compared to my local machine which is failing very differently with 2,564 errors all like 
 `C:\Dev\Should\src\Should.Core\Assertions\Assertions.cs(465,23): error CS0518: Predefined type 'System.Object' is not defined or imported [C:\Dev\Shou
ld\src\Should-netcore45\Should-netCore45.csproj]`

At the moment I'm tempted to wipe and rebuild my machine, so confirmation that isn't the problem would be great :-)

alski commented:
P.S. If this PR14 is now closed, shall we continue the conversation on https://github.com/erichexter/Should/pull/11 ?
